Build fails on arm-none-eabi-gcc 11.1.0

Followed the instructions from the readme. But my build fails. My gcc version is
```
arm-none-eabi-gcc (Arch Repository) 11.1.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I get the following error during the build process. Full log attached. 

```
In function 'text_screen_allow',
    inlined from 'text_screen_init' at ../../../../libtock/text_screen.c:50:11:
../../../../libtock/text_screen.c:37:28: error: '<unknown>' may be used uninitialized [-Werror=maybe-uninitialized]
   37 |   allow_ro_return_t aval = allow_readonly(DRIVER_NUM_TEXT_SCREEN, 0, ptr, size);
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
[log.txt](https://github.com/tock/libtock-c/files/6768987/log.txt)

--- a/libtock/text_screen.c
+++ b/libtock/text_screen.c
@@ -34,7 +34,10 @@ static int text_screen_command (int command_num, int data1, int data2) {
 }
 
 static int text_screen_allow (const void* ptr, size_t size) {
+  #pragma GCC diagnostic push
+  #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
   allow_ro_return_t aval = allow_readonly(DRIVER_NUM_TEXT_SCREEN, 0, ptr, size);
+  #pragma GCC diagnostic pop
   return tock_allow_ro_return_to_returncode(aval);
 }
 

